scheduler/filter/slots.go

```diff
@@ -0,0 +1,48 @@
+package filter
+
+import (
+	"errors"
+	"github.com/docker/swarm/cluster"
+	"github.com/docker/swarm/scheduler/node"
+	"strconv"
+)
+
+var (
+	// ErrNoNodeWithFreeSlotsAvailable is exported
+	ErrNoNodeWithFreeSlotsAvailable = errors.New("No node with enough open slots available in the cluster")
+)
+
+//SlotsFilter only schedules containers with open slots.
+type SlotsFilter struct {
+}
+
+// Name returns the name of the filter
+func (f *SlotsFilter) Name() string {
+	return "containerslots"
+}
+
+// Filter is exported
+func (f *SlotsFilter) Filter(_ *cluster.ContainerConfig, nodes []*node.Node, _ bool) ([]*node.Node, error) {
+	result := []*node.Node{}
+
+	for _, node := range nodes {
+
+		if slotsString, ok := node.Labels["containerslots"]; ok {
+			slots, err := strconv.Atoi(slotsString) //if err => cannot cast to int, so no candidate for us
```

Comment:
I'm thinking about 2 different approaches.
- This label is invalid so it doesn't take effect. The node would be included as a candidate. You may add a error logging for the manager of this invalid label. It might generate too much log if we do it here though. 
- Do validation on this label and put the node to `unhealthy` state. Nodes in `unhealthy` state wouldn't take new containers. But `docker -H manager:port info` would show the node as `unhealthy`. So it's an explicit failure. 

What do you think?